middlewares/verifyToken.js

```diff
@@ -0,0 +1,29 @@
+import jwt from 'jsonwebtoken';
+
+const verifyToken = (req, res, next) => {
+  const token = req.headers.authorization.split(' ')[1];
+  if (!token) {
+    return res.status(401).json({
+      success: false,
+      errors: {
+        body: ['No token has been provided in the request'],
+      }
+    });
+  }
+  jwt.verify(token, process.env.JWT_TOKEN_SECRET, (err, decoded) => {
+    if (err) {
+      return res.status(401).json({
+        success: false,
+        errors: {
+          body: ['Could not authenticate the provided token'],
+        }
+      });
+    }
+    req.userId = decoded.id;
```

Comment:
What do you think about using the 'decode.id' to make the user object available in this middleware.

Reply:
That is why I reassigned it to req.userId, to make the id from the token available in the request for the next method in line to make use of.

Reply:
I think what I need to do is create another middleware that retrieves the user details and attaches it to the request object instead of doing that in this middleware all because of the fact that it's not all the endpoint we use this middleware on that needs to hit the db for that specific user info. So the new middleware that will be created can be used only when that is absolutely needed.

Reply:
That also makes good sense, cheers.

Reply:
Done